yt_dlp/YoutubeDL.py

```diff
@@ -3171,6 +3171,8 @@ def correct_ext(filename, ext=new_ext):
                                 'f%s' % f['format_id'], info_dict['ext'])
                             downloaded.append(fname)
                         info_dict['url'] = '\n'.join(f['url'] for f in requested_formats)
+                        info_dict['http_headers'] = traverse_obj(
+                            info_dict, 'http_headers', ('requested_formats', ..., 'http_headers'), get_all=False)
```

Comment:
This should be handled in FFmpegFD imo since the two formats could need different headers in theory